crypto/pkcs11/pkcs11helpers.go

```diff
@@ -68,7 +68,23 @@ type Pkcs11KeyFile struct {
 
 // Pkcs11KeyFileObject is a representation of the Pkcs11KeyFile with the pkcs11 URI as an object
 type Pkcs11KeyFileObject struct {
-	uri pkcs11uri.Pkcs11URI
+	Uri *pkcs11uri.Pkcs11URI
```

Comment:
I just looked at the library interfaces. I would be comfortable if the library had explicit YAML serialization/deserialization interfaces. If not there seems to be some implicit contract on how to serialize/deserialize this into YAML.

Reply:
I am converting the YAML file to this object which can also carry the environment variables as a (recently added) 'convenience'. The other way would be to wrap the Pkcs11URI into a Pkcs11YAML object and add a map for the environment variables there rather than the one of the Pkcs11URI. The Pkcs11URI carries the pkcs11 module name (`...&module-name=softhsm2) and so I thought it would be convenient to have the environment variables there as well that help use the module.

There is no conversion of the pkcs11uri.Pkcs11URI back to yaml although there could be.

So what do you concretely suggest?

Reply:
Oh, we don't have environment variables as part of the YAML in this PR, yet. The next PR, however, would introduce this and then convert the public key test that uses the plain URI to use a YAML file instead since this is more expressive.

Reply:
Ok let's take a look at this when we introduce that, good for now.